components/reddit/reddit.app.mjs

```diff
@@ -1,27 +1,123 @@
+          get(subreddit, "value", subreddit),
+          params,
+        );
+        const options = get(links, "data.children", []).map((item) => ({
+          label: item.data.title,
+          value: {
+            label: item.data.title,
+            value: item.data.id,
+            name: item.data.name,
+          },
+        }));
+
+        return {
+          options,
+          context: {
+            after: options.length > 0 && get(options, `${options.length - 1}.value.name`),
+          },
+        };
+      },
+    },
+    after: {
+      type: "string",
+      label: "After",
+      description: "Only one of `after` and `before` should be specified. These indicate the [fullname](https://www.reddit.com/dev/api/#fullnames) of an item in the listing to use as the anchor point of the slice.",
+      optional: true,
+    },
+    before: {
+      type: "string",
+      label: "Before",
+      description: "Only one of `after` and `before` should be specified. These indicate the [fullname](https://www.reddit.com/dev/api/#fullnames) of an item in the listing to use as the anchor point of the slice.",
+      optional: true,
+    },
+    count: {
+      type: "integer",
+      label: "Count",
+      description: "The number of items already seen in this listing. On the html site, the builder uses this to determine when to give values for `before` and `after` in the response.",
+      optional: true,
+    },
+    limit: {
+      type: "integer",
+      label: "Limit",
+      description: "Default to 25. The maximum number of items desired",
+      min: 1,
+      max: 100,
+      optional: true,
+    },
     subreddit: {
       type: "string",
       label: "Subreddit",
       description: "The subreddit you'd like to watch.",
       useQuery: true,
-      async options(context) {
-        const q = context.query;
-        const options = [];
-        const results = await this.getAllSearchSubredditsResults(q);
-        for (const subreddit of results) {
-          options.push({
-            label: subreddit.title,
-            value: subreddit.displayName,
-          });
-        }
-        return options;
+      async options({
+        query,
+        prevContext,
+      }) {
+        const res = await this.searchSubreddits({
+          after: prevContext?.after,
+          q: query,
+          limit: 30,
+          show_users: false,
+          sort: "relevance",
+          sr_detail: false,
+          typeahead_active: false,
+        });
+
+        const subreddits = get(res, "data.children", []);
+        const options = subreddits.map((subreddit) => ({
+          label: subreddit.data.title,
+          value: {
+            label: subreddit.data.title,
+            value: subreddit.data.displayName,
+            name: subreddit.data.name,
+          },
+        }));
+
+        return {
+          options,
+          context: {
+            after: options.length > 0 && get(options, `${options.length - 1}.value.name`),
```

Comment:
**-Minor-**
I think the `options.length > 0` check isn't necessary here because [`_.get`](https://lodash.com/docs/4.17.15#get) will return `undefined` if `${options.length - 1}` is `-1`.

```suggestion
            after: get(options, `${options.length - 1}.value.name`),
```